src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -416,7 +418,7 @@ private boolean tryToLaunchJournalNode(SchedulerDriver driver, Offer offer) {
     log.info(deadJournalNodes);
 
     if (deadJournalNodes.isEmpty()) {
-      if (liveState.getJournalNodeSize() == conf.getJournalNodeCount()) {
+      if (persistentState.getJournalNodes().size() == conf.getJournalNodeCount()) {
```

Comment:
For these getters we could cache the state locally, so that we only have to pull from zk on scheduler failover/recovery. We'll still push updates out to ZK asap so that the state is up to date in the event of failover. This is probably worth a separate issue/PR.